src/openforms/contrib/zgw/service.py

```diff
@@ -84,6 +84,11 @@ def create_attachment_document(
             status="definitief",
             filename=submission_attachment.get_display_name(),
             description="Bijgevoegd document",
+            received_date=(
+                options["ontvangstdatum"].date().isoformat()
+                if options["ontvangstdatum"]
+                else None
+            ),
```

Comment:
To be consistent with the other `create_document` args, why not do `received_date=options.get("ontvangstdatum")` here, and set the correct date before like so:

```python
if submission.is_completed:
    attachment_options["ontvangstdatum"] = submission.completed_on.date().isoformat()
```

I'm also wondering if this check is necessary, as submissions should be completed when registration plugins are called. I think it's better to assume `submission.completed_on` is set, so that if it isn't it crashes and at least we know something's wrong

Reply:
I agree that during the registration the submission should be completed https://github.com/open-formulieren/open-forms/blob/master/src/openforms/registrations/tasks.py#L166 but I have to investigate why these tests are failing. They are not if I add an if-else statement and set `ontvangstdatum` either to a string or `None`

Reply:
`completed_on` is set when the `/api/v2/submissions/:uuid/_complete` endpoint succeeds, and that kicks off the celery tasks in the background that run the registration.

There is a factory trait to make sure this is set in a consistent way, probably the existing tests were written before this trait existed or they weren't aware of it at the time of writing the test/feature.

Feel free to just write the registration plugin code as `submission.completed_on.isoformat()` - any failing tests need to have their initial state fixed. Registering a submission that isn't completed does not make sense and should crash hard.